tests/Tests/Protocols/Wishbone.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ApplicativeDo       #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE NumericUnderscores  #-}
+{-# LANGUAGE RecordWildCards     #-}
+-- The exhaustiveness-checker doesn't work well with ViewPatterns
+{-# OPTIONS_GHC -Wno-incomplete-patterns #-}
+module Tests.Protocols.Wishbone where
+
+
+import           Clash.Prelude                  hiding ((&&))
+
+import           Clash.Hedgehog.Sized.BitVector
+
+import           Hedgehog
+
+import           Hedgehog.Gen                   as Gen
+
+import           Hedgehog.Range                 as Range
+
+import           Protocols
+import           Protocols.Hedgehog             (defExpectOptions)
+import           Protocols.Wishbone
+import           Protocols.Wishbone.Hedgehog
+
+-- tasty
+import           Control.DeepSeq                (NFData)
+import           Data.Either                    (isLeft)
+import           Prelude                        hiding (undefined)
+import           Test.Tasty
+import           Test.Tasty.Hedgehog            (HedgehogTestLimit (HedgehogTestLimit))
+import           Test.Tasty.Hedgehog.Extra      (testProperty)
+import           Test.Tasty.TH                  (testGroupGenerator)
+
+smallInt :: Range Int
+smallInt = Range.linear 0 10
+
+genSmallInt :: Gen Int
+genSmallInt = Gen.integral smallInt
+
+genData :: Gen a -> Gen [a]
+genData = Gen.list (Range.linear 0 150)
+
+genWishboneTransfer
+  :: (KnownNat addressWidth)
+  => Gen a
+  -> Gen (WishboneMasterRequest addressWidth a)
+genWishboneTransfer genA = Gen.choice
+  [ Read <$> genDefinedBitVector
+  , Write <$> genDefinedBitVector <*> genA ]
+
+--
+-- 'id' circuit
+--
+
+idWriteWbSt
```

Comment:
that might be a better name! However I am thinking about what to do with this test case in general because it actually doesn't pass the common-sense rules (writes with ack and no data only, reads with ack and valid data)